docs/filmfestival-sql.py

```diff
@@ -0,0 +1,113 @@
+festival_template = (
+    "<html><head><title>Just some arbitrary text</title></head>"
+    '<body><h1 style="text-align:center">Hook Norton Film Festival</h1>'
+    "<ol>"
+    '<li id="filmtemplate">'
+    '<b id="filmtitle"></b>'
+    "<dl>"
+    '<dt>Director<dd id="director">'
+    '<dt>Release Year<dd id="filmyear">'
+    '<dt>Cast<dd id="cast">'
+    "</dl>"
+    "</li>"
+    "</ol>"
+    "</body></html"
+)
+
+# -------------------------------------------------------------------
+# define database access
+# -------------------------------------------------------------------
+dbfilename = __file__.replace(".py", ".db")  # the SQLITE database file name
+database = sqlite3.connect(dbfilename)  # open database
+cursor_films = database.cursor()  # cursor for selecting the films
+cursor_casts = database.cursor()  # cursor for selecting actors per film
+
+# select statement for the films - let SQL also sort it for us
+select_films = """SELECT title, director, year FROM films ORDER BY title"""
+
+# select stament for actors, a skeleton: sub-select by film title
+select_casts = """SELECT name FROM actors WHERE film = "%s" ORDER BY name"""
+
+# -------------------------------------------------------------------
+# define the HTML Story and fill it with database data
+# -------------------------------------------------------------------
+story = fitz.Story(festival_template)
+body = story.body  # access the HTML body detail
+template = body.find(None, "id", "filmtemplate")  # find the template part
+
+# read the films from the database and put them all in one Python list
+# NOTE: instead we might fetch rows one by one (advisable for large volumes)
+cursor_films.execute(select_films)  # execute cursor, and ...
+films = cursor_films.fetchall()  # read out what was found
+
+for title, director, year in films:  # iterate through the films
+    film = template.clone()  # clone template to report each film
+    film.find(None, "id", "filmtitle").add_text(title)  # put title in templ
+    film.find(None, "id", "director").add_text(director)  # put director
+    film.find(None, "id", "filmyear").add_text(str(year))  # put year
+
+    # the actors reside in their own table - find the ones for this film title
+    cursor_casts.execute(select_casts % title)  # execute cursor
+    casts = cursor_casts.fetchall()  # read actors for the film
+    # each actor name appears in its own tuple, so extract it from there
+    film.find(None, "id", "cast").add_text("\n".join([c[0] for c in casts]))
+    body.append_child(film)
+
+template.remove()  # remove the template
+
+# -------------------------------------------------------------------
+# generate the PDF
+# -------------------------------------------------------------------
+writer = fitz.DocumentWriter(__file__.replace(".py", ".pdf"), "compress")
+mediabox = fitz.paper_rect("a4")  # use pages in ISO-A4 format
+where = mediabox + (72, 36, -36, -72)  # leave page borders
+
+done = 1  # end of output indicator
```

Comment:
I think this variable should be called `more`, and the code should look like:
```
more = 1
while more:
    dev = ...
    more, filled = story.place(where)
    ....
```
[I need to change `fitz/fitz.i:place()` to call its internal return variable `more` too.]

Reply:
You are right - I will update this.

Reply:
Great, thanks. There are some other similar occurrences later on - would you be ok to update them also?